Sources/NIOCore/Docs.docc/swift-concurrency.md

```diff
@@ -0,0 +1,244 @@
+return the finished `NIOProtocolNegotiationResult`.
+```swift
+let serverBoostrap = try await ServerBootstrap(group: eventLoopGroup)
+    .childChannelInitializer { channel in
+        channel.eventLoop.makeCompletedFuture {
+            let negotiationHandler = NIOTypedApplicationProtocolNegotiationHandler<NegotiationResult>(eventLoop: channel.eventLoop) { alpnResult, channel in
+                switch alpnResult {
+                case .negotiated(let alpn):
+                    switch alpn {
+                    case "string":
+                        return channel.eventLoop.makeCompletedFuture {
+                            let asyncChannel = try NIOAsyncChannel(
+                                synchronouslyWrapping: channel,
+                                isOutboundHalfClosureEnabled: true,
+                                inboundType: String.self,
+                                outboundType: String.self
+                            )
+
+                            return NIOProtocolNegotiationResult.finished(NegotiationResult.string(asyncChannel))
+                        }
+                    case "byte":
+                        return channel.eventLoop.makeCompletedFuture {
+                            let asyncChannel = try NIOAsyncChannel(
+                                synchronouslyWrapping: channel,
+                                isOutboundHalfClosureEnabled: true,
+                                inboundType: UInt8.self,
+                                outboundType: UInt8.self
+                            )
+
+                            return NIOProtocolNegotiationResult.finished(NegotiationResult.byte(asyncChannel))
+                        }
+                    default:
+                        return channel.eventLoop.makeFailedFuture(ProtocolNegotiationError())
+                    }
+                case .fallback:
+                    return channel.eventLoop.makeFailedFuture(ProtocolNegotiationError())
+                }
+            }
+
+            try channel.pipeline.syncOperations.addHandler(negotiationHandler)
+        }
+    }
+```
+
+Lastly, we can now bind the `serverChannel` and handle the incoming connections. In the code below,
+you can see that our server channel is now a `NIOAsyncChannel` of `NegotiationResult`s instead of
+child channels.
+```swift
+let serverChannel = serverBootstrap.bind(
+    host: "127.0.0.1",
+    port: 1995,
+    protocolNegotiationHandlerType: NIOTypedApplicationProtocolNegotiationHandler<NegotiationResult>.self
+)
+
+try await withThrowingDiscardingTaskGroup { group in
+    for try await negotiationResult in serverChannel.inboundStream {
+        group.addTask {
+            do {
+                switch connectionChannel {
+                case .string(let channel):
+                    for try await value in channel.inboundStream {
+                        continuation.yield(.string(value))
```

Comment:
Where did the continuation come from? Is this meant to be echo-ing back the value instead?